File: DeliveryBot/DeliveryBot.Db/Models/Delivery.cs
namespace DeliveryBot.Db.Models;

public class Delivery : Entity
{
    public DateTime ShippedDateTime { get; set; }
    public DateTime DeliveredDateTime { get; set; }

    public Guid? RobotId { get; set; }
    public Robot? Robot { get; set; }

    public Guid? CompanyEmployeeId { get; set; }
    public CompanyEmployee? CompanyEmployee { get; set; }
    
    public Guid? OrderId { get; set; }
    public Order? Order { get; set; }
}

File: DeliveryBot/deliverybot.client/src/interfaces/delivery.ts
import { GetDeliveryRobotQueryDto } from "./robot";

export interface DeliveryInfoDto {
  shippedDateTime?: Date | undefined;
  deliveredDateTime?: Date | undefined;
  robotId?: string | undefined;
  orderId?: string | undefined;
  id?: string;
}

export interface DeliveryFullInfo extends DeliveryInfoDto {
  robot: GetDeliveryRobotQueryDto
}

export interface CreateDeliveryCommand {
  shippedDateTime?: Date;
  robotId?: string | undefined;
  orderId?: string | undefined;
}

